storybookjs/storybook
# Build: Migrate `@storybook/html-vite` to strict TS

## What I did

Part of #22176 migration to type-strictness for `@storybook/html-vite`:

1. Change in tsconfig.json "strict": false -> "strict": true
2. Run yarn check in the directory of the package.
3. Fix all TS errors.

## How to test
0. run `yarn task --task compile --no-link --start-from=install`
1. cd into `code/frameworks/html-vite`
2.`yarn check` from project root
3. Run `yarn task --task dev --template html-vite/default-ts --start-from=auto` to check everything still builds in sandbox
4. Run e2e tests against dev sandbox `yarn task --task e2e-tests-dev --template html-vite/default-ts` 
5. Run `yarn run test` from root of codebase
6. Run `yarn run test:ci` from root of codebase, drum fingers.... and wait for checking validity types ` nx run @storybook/html-vite:check` to succeed.
 
## Checklist



- [ ] Make sure to add/update documentation regarding your changes -- n/a
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

      make sure to add the `ci:merged` or `ci:daily` GH label to it.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`
